plugin/src/main/java/at/helpch/chatchat/command/MainCommand.java

```diff
@@ -2,34 +2,19 @@
 
 import at.helpch.chatchat.ChatChatPlugin;
 import at.helpch.chatchat.api.User;
+import at.helpch.chatchat.util.MessageUtils;
 import dev.triumphteam.cmd.core.annotation.Default;
 import net.kyori.adventure.text.Component;
-import net.kyori.adventure.text.event.ClickEvent;
-import net.kyori.adventure.text.format.NamedTextColor;
-import net.kyori.adventure.text.format.TextColor;
-import org.jetbrains.annotations.NotNull;
-
-import static net.kyori.adventure.text.Component.text;
+import org.bukkit.plugin.java.JavaPlugin;
 
 public final class MainCommand extends ChatChatCommand {
 
-    private final ChatChatPlugin plugin;
-
-    public MainCommand(@NotNull final ChatChatPlugin plugin) {
-        this.plugin = plugin;
-    }
+    private static final JavaPlugin PLUGIN = JavaPlugin.getProvidingPlugin(ChatChatPlugin.class);
```

Comment:
would it make sense to do this everywhere instead of injecting?

Reply:
no, that would be considered bad practice. I only use it here to make TEXT a constant (because it doesnt need be associated with an object). JavaPlugin#getPlugin/getProvidingPlugin should only be used where necessary

Reply:
i see, thanks